agents/src/llm/chat_context.ts

```diff
@@ -0,0 +1,105 @@
+// SPDX-FileCopyrightText: 2024 LiveKit, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+import type { AudioFrame } from '@livekit/rtc-node';
+
+export enum ChatRole {
+  SYSTEM,
+  USER,
+  ASSISTANT,
+  TOOL,
+}
+
+export interface ChatImage {
+  image: string | AudioFrame;
+  inferenceWidth?: number;
+  inferenceHeight?: number;
+  /** Used by LLM implementations to store a processed version of the image for later use. */
+  cache: { [id: string | number | symbol]: any };
```

Comment:
any way to make sure "cache" is private?

Reply:
to what degree? LLM needs direct access to it, so we can't make it private or readonly. i can mark is as `@internal` so it's clear only llms should use it

Reply:
internal sounds good